src/highlight.rs

```diff
@@ -46,7 +48,7 @@ impl Boundary {
     }
 }
 
-fn replace_tabs(s: &str, tab_len: u8) -> Cow<'_, str> {
+fn replace_tabs(s: &str, tab_len: u8, mask: bool) -> Cow<'_, str> {
```

Comment:
```suggestion
fn replace_tabs(s: &str, tab_len: u8, mask: bool) -> Cow<'_, str> {
    if mask {
        return Cow::Owned(s.chars().map(|_| '*').collect());
    }
```

I think tab character should be always counted as one character because `*` means a single character. So it should be replaced before replacing tab characters instead of converting it at the end of this method.

Reply:
Thats really very hard to do. It works in hard tab mode because the \t gets passed to replace_tabs. But in soft tab mode (the default) the replacement is done into the text buffer  (`insert_tab` function) before it reaches the highlighter

Counter argument , to keep as it is. When I type 'tab' your code types n spaces for me, those n spaces become part of the output thats returned (in soft mode), so that what should get masked in the display

Counter counter argument. This code will be used 99% of the time to enter passwords, no password will ever have \t in it since its impossible to enter into a gui. So it really doesnt matter. In the 1% case the user will quickly work out what is happening

Reply:
Sorry my comment was unclear. As you said, soft tab mode means whitespaces are inserted when user input tab character. So it is not possible. I think we can consider hard tabs only.
I believe almost all users of masking text feature can simply disable soft tabs by `textarea.set_hard_tab_indent(true)`.